common/src/main/java/org/apache/celeborn/common/network/protocol/Message.java

```diff
@@ -150,12 +154,113 @@ public static Type decode(ByteBuf buf) {
           throw new IllegalArgumentException("Unknown message type: " + id);
       }
     }
+
+    public static Type decode(ByteBuf buf) {
+      byte id = buf.readByte();
+      switch (id) {
+        case 0:
+          return CHUNK_FETCH_REQUEST;
+        case 1:
+          return CHUNK_FETCH_SUCCESS;
+        case 2:
+          return CHUNK_FETCH_FAILURE;
+        case 3:
+          return RPC_REQUEST;
+        case 4:
+          return RPC_RESPONSE;
+        case 5:
+          return RPC_FAILURE;
+        case 6:
+          return OPEN_STREAM;
+        case 7:
+          return STREAM_HANDLE;
+        case 9:
+          return ONE_WAY_MESSAGE;
+        case 11:
+          return PUSH_DATA;
+        case 12:
+          return PUSH_MERGED_DATA;
+        case 13:
+          return REGION_START;
+        case 14:
+          return REGION_FINISH;
+        case 15:
+          return PUSH_DATA_HAND_SHAKE;
+        case 16:
+          return READ_ADD_CREDIT;
+          //        case 17:
+          //          return READ_DATA;
+        case -1:
+          throw new IllegalArgumentException("User type messages cannot be decoded.");
+        default:
+          throw new IllegalArgumentException("Unknown message type: " + id);
+      }
+    }
   }
 
   public static Message decode(Type msgType, ByteBuf in) {
     return decode(msgType, in, true);
   }
 
+  public static Message decode(Type msgType, ByteBuffer in) {
+    return decode(msgType, in, true);
+  }
+
+  public static Message decode(Type msgType, ByteBuffer in, boolean decodeBody) {
+    switch (msgType) {
+      case CHUNK_FETCH_REQUEST:
+        return ChunkFetchRequest.decode(in);
+
+      case CHUNK_FETCH_SUCCESS:
+        return ChunkFetchSuccess.decode(in, decodeBody);
+
+      case CHUNK_FETCH_FAILURE:
+        return ChunkFetchFailure.decode(in);
+
+      case RPC_REQUEST:
+        return RpcRequest.decode(in, decodeBody);
+
+      case RPC_RESPONSE:
+        return RpcResponse.decode(in, decodeBody);
+
+      case RPC_FAILURE:
+        return RpcFailure.decode(in);
+
+      case OPEN_STREAM:
+        return OpenStream.decode(in);
+
+      case STREAM_HANDLE:
+        return StreamHandle.decode(in);
+
+      case ONE_WAY_MESSAGE:
+        return OneWayMessage.decode(in, decodeBody);
+
+      case PUSH_DATA:
+        return PushData.decode(in, decodeBody);
+
+      case PUSH_MERGED_DATA:
+        return PushMergedData.decode(in, decodeBody);
+
+      case REGION_START:
+        return RegionStart.decode(in);
+
+      case REGION_FINISH:
+        return RegionFinish.decode(in);
+
+      case PUSH_DATA_HAND_SHAKE:
+        return PushDataHandShake.decode(in);
+
+      case READ_ADD_CREDIT:
+        return ReadAddCredit.decode(in);
```

Comment:
now , rpc and ReadAddCredit add decode method and others messages doesn't need add this function

Reply:
OK, I already removed unnecessary changes.